CodePush.m

```diff
@@ -505,7 +506,12 @@ - (void)savePendingUpdate:(NSString *)packageHash
 RCT_EXPORT_METHOD(notifyApplicationReady:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    [CodePush removePendingUpdate];
+    // We only mark a pending update as succeeded only if it has been booted up
+    // successfully, during which `_isFirstRunAfterUpdate` is set to true.
+    if (_isFirstRunAfterUpdate) {
```

Comment:
That way I can completely undo this change. Plus it could simplify the reportStatus logic too.